docs/products/opensearch/howto/opensearch-search-and-python.rst

```diff
@@ -0,0 +1,471 @@
+- `index.py <https://github.com/aiven/demo-opensearch-python/blob/main/index.py>`_, contains methods that manipulate the index
+- `search.py <https://github.com/aiven/demo-opensearch-python/blob/main/search.py>`_, contains search queries methods
+- `helpers.js <https://github.com/aiven/demo-opensearch-python/blob/main/helpers.py>`_, response handler of search requests
+
+We will be using ``Typer`` `Python library <ttps://typer.tiangolo.com/>`_ to create CLI commands to run from the terminal.
+
+Connect to the OpenSearch cluster in Python
+-------------------------------------------
+
+Follow instructions on how to :doc:`connect to the cluster with a Python client <connect-with-python>` and add the necessary code to ``config.py``.
+In this ``config.py``, the cluster configuration is created. After this, we're set to interact with our cluster.
+
+.. code:: python
+
+    import os
+
+    from dotenv import load_dotenv
+    from opensearchpy import OpenSearch
+
+
+    load_dotenv()
+    INDEX_NAME = "epicurious-recipes"
+    SERVICE_URI = os.getenv("SERVICE_URI")
+    client = OpenSearch(SERVICE_URI, use_ssl=True)
+
+
+.. tip::
+
+    The ``SERVICE_URI`` value can be found in the Aiven Console dashboard.
+
+Upload data to OpenSearch using Python
+--------------------------------------
+
+Once you're connected, the next step should be to inject data into our cluster. Follow the instructions in how to :ref:`load the sample data <load-data-with-python>` into your OpenSearch cluster. 
+
+You can run from the demo following::
+
+  python index.py send-data
+
+Once the data is sent, we can :ref:`retrieve the data mapping <get-mapping-with-python>` to explore the structure of the data, with their respective fields and types. 
+
+From the `demo <https://github.com/aiven/demo-opensearch-python/blob/main/index.py#L50>`_, you can check the data mapping running::
+
+  python index.py get-mapping
+
+
+All set to start writing your own search queries.
+
+Query the data
+''''''''''''''
+
+``search()`` method
+-------------------
+
+We have an OpenSearch client and data in your cluster, so we can start writing search queries. The method ``search()`` provided by the Python OpenSearch client library will be used to perform the queries. 
+
+Find below the method signature's::
+
+  client.search: (body=None, index=None, doc_type=None, params=None, headers=None)
+
+All the parameters are optional in the ``search()`` method. We will define Python ``dict()`` objects for the ``body`` parameter and use. Here is an example:
+
+.. code:: python
+   query_body = {
+                  "query": {
+                    "match_all": {}
+                  }
+                }
+
+
+The parameters that we will be using to configure our search queries are the ``index`` and ``body``. The ``index`` will not be changing, but we will be modifying the ``body`` parameter value according to the query purpose.
```

Comment:
an idea - maybe instead of "The ``index`` will not be changing" say that the index parameter is the name of the index we used to load the data and therefore it won't be changed. In this way you help to understand the reason behind this.